src/Microsoft.ComponentDetection.Detectors/gradle/GradleComponentDetector.cs

```diff
@@ -22,6 +28,17 @@ public class GradleComponentDetector : FileComponentDetector, IComponentDetector
         this.ComponentStreamEnumerableFactory = componentStreamEnumerableFactory;
         this.Scanner = walkerFactory;
         this.Logger = logger;
+        var envVar = Environment.GetEnvironmentVariable(ProdDepsEnvVar);
```

Comment:
we have a `IEnvironmentVariableService` to allow mocking and testing of env variables

Reply:
lets also update the tests to not set environment variables as well